app/components/markdown/markdown_image/index.tsx

```diff
@@ -88,7 +88,7 @@ const MarkdownImage = ({
     const uri = source.startsWith('/') ? serverUrl + source : source;
 
     const fileInfo = useMemo(() => {
-        const link = decodeURIComponent(uri);
+        const link = safeDecodeURIComponent(uri);
```

Comment:
Should we test this?

An image file with `hello%253Fworld.jpg`.  Would it get decoded into `hello%3Fworld.jpg` thus causing the file not to be found?